### ISSUE TITLE
TypeError: Cannot convert input to Timestamp

### ISSUE THREAD

**Issue by cbogithub** [enigmampc/catalyst#167]
Dear Catalyst Maintainers,

Before I tell you about my issue, let me describe my environment:
Environment

    Operating System: (Windows Version or $ uname --all)
    window 10

    Python Version: $ python --version
    python3.6

    Python Bitness: $ python -c 'import math, sys;print(int(math.log(sys.maxsize + 1, 2) + 1))'
    x64

    How did you install Catalyst: (pip, conda, or other (please explain))

    Python packages: $ pip freeze or $ conda list
    aiodns==1.1.1
    aiohttp==2.3.7
    alabaster==0.7.7
    alembic==0.9.6
    aniso8601==1.0.0
    asciichart==0.1
    asciichartpy==1.5.7
    asn1crypto==0.24.0
    astroid==1.5.3
    async-timeout==2.0.0
    attrs==17.4.0
    autobahn==17.10.1
    automat==0.6.0
    Babel==2.2.0
    banal==0.3.3
    bcolz==0.12.1
    beautifulsoup4==4.6.0
    bleach==1.5.0
    blinker==1.4
    bokeh==0.12.13
    boto3==1.5.14
    botocore==1.8.28
    Bottleneck==1.2.1
    browsermob-proxy==0.8.0
    cachepy==0.1
    cchardet==2.1.1
    ccxt==1.10.620
    certifi==2017.11.5
    cffi==1.11.2
    chardet==3.0.4
    click==6.6
    colorama==0.3.7
    constantly==15.1.0
    contextlib2==0.5.5
    coverage==4.0.3
    cryptography==2.1.4
    cx-freeze==5.0
    cycler==0.10.0
    cymem==1.31.2
    cyordereddict==1.0.0
    Cython==0.27.3
    cytoolz==0.8.2
    dash==0.19.0
    dash-core-components==0.17.0
    dash-html-components==0.8.0
    dash-renderer==0.11.1
    dataset==1.0.5
    decorator==4.1.2
    defusedxml==0.4.1
    dill==0.2.7.1
    Django==1.10.2
    django-autocomplete-light==3.2.0
    django-extensions==1.7.4
    django-gm2m==0.4.7
    django-querysetsequence==0.7
    django-tagging==0.4.5
    django-taggit==0.21.3
    docutils==0.12
    dropbox==6.4.0
    ecdsa==0.13
    empyrical==0.2.2
    enum34==1.1.6
    facebook-sdk==2.0.0
    fbchat==1.0.19
    Flask==0.12.2
    Flask-API==0.6.6.post1
    Flask-AppBuilder==1.7.0
    Flask-Auth==0.85
    Flask-BabelPkg==0.9.6
    Flask-BasicAuth==0.2.0
    Flask-Bootstrap3==3.1.1.3
    Flask-Compress==1.4.0
    Flask-Cors==2.1.0
    Flask-Dropbox==0.3
    Flask-Excel==0.0.4
    Flask-GoogleAuth==0.4.2
    Flask-GoogleMaps==0.1.9
    Flask-Headers==1.0
    Flask-HTTPAuth==3.1.2
    Flask-HTTPS==0.1.0
    Flask-JSON==0.3.0
    Flask-Jsonpify==1.5.0
    Flask-JWT==0.3.2
    Flask-Login==0.2.11
    Flask-Mail==0.9.1
    Flask-Navigation==0.2.0
    Flask-OpenID==1.2.5
    Flask-Principal==0.4.0
    Flask-RESTful==0.3.4
    flask-secure-cookie==0.1.2
    flask-secure-headers==0.6
    Flask-Security==1.7.5
    Flask-ServiceLayer==0.0.4
    Flask-Session==0.2.3
    Flask-SocketIO==2.7.1
    Flask-SQLAlchemy==2.0
    Flask-WTF==0.12
    ftfy==4.4.3
    future==0.15.2
    geopy==1.11.0
    gevent==1.1.2
    google-api-python-client==1.6.3
    gpsoauth==0.4.0
    greenlet==0.4.5
    html5lib==0.9999999
    httplib2==0.10.3
    hyperlink==17.3.1
    idna==2.6
    imutils==0.4.3
    incremental==17.5.0
    intervaltree==2.1.0
    IPy==0.83
    ipython-genutils==0.2.0
    isort==4.2.15
    itsdangerous==0.24
    Jinja2==2.8
    jmespath==0.9.3
    joblib==0.11
    jsonschema==2.6.0
    jupyter-core==4.4.0
    Keras==2.0.8
    Kivy-Garden==0.1.4
    lazy-object-proxy==1.3.1
    logbook==1.1.0
    lru-dict==1.1.6
    lxml==4.1.1
    Mako==1.0.7
    Markdown==2.6.9
    MarkupSafe==0.23
    matplotlib==2.1.0
    mccabe==0.6.1
    MechanicalSoup==0.9.0.post4
    mock==2.0.0
    multidict==3.3.2
    multipledispatch==0.4.9
    murmurhash==0.26.4
    nbformat==4.4.0
    networkx==2.0
    nltk==3.2.5
    normality==0.5.7
    nose==1.3.7
    Nuitka64==5.1.190
    numexpr==2.6.4
    numpy==1.13.3+mkl
    oauth2client==4.1.2
    olefile==0.44
    opencv-contrib-python==3.3.0.9
    opencv-python==3.3.0.9
    pandas==0.18.1
    pandas-datareader==0.5.0
    paramiko==1.16.0
    passlib==1.6.5
    pathlib==1.0.1
    patsy==0.5.0
    pbr==1.10.0
    peewee==2.8.1
    picamera==1.12
    Pillow==4.3.0
    plac==0.9.6
    plotly==2.2.3
    pluggy==0.3.1
    preshed==1.0.0
    protobuf==3.4.0
    protobuf-to-dict==0.1.0
    py==1.4.34
    py2exe==0.9.2.0
    pyasn1==0.4.2
    pyasn1-modules==0.2.1
    pycares==2.3.0
    pycparser==2.18
    pycryptodome==3.4.6
    pycurl==7.19.3.1
    pyexcel==0.2.2
    pyexcel-io==0.2.0
    pyexcel-webio==0.0.7
    Pygments==2.1.1
    PyJWT==1.4.0
    PyLD==0.7.1
    pylint==1.7.4
    pymessenger==0.0.7.0
    pymongo==3.5.1
    PyOpenGL==3.1.0
    PyOpenGL-accelerate==3.1.0
    pyOpenSSL==17.5.0
    pyparsing==2.2.0
    pypiwin32==220
    pypng==0.0.18
    pyqrcode==1.2.1
    PyQt5==5.9
    pyreadline==2.1
    pytest==3.2.2
    pytest-cov==2.4.0
    pytest-django==3.0.0
    python-binance==0.5.17
    python-crfsuite==0.9.5
    python-dateutil==2.6.1
    python-editor==1.0.3
    python-engineio==1.0.3
    python-socketio==1.6.0
    python-telegram-bot==8.1.1
    python-wordpress-xmlrpc==2.3
    python3-openid==3.0.10
    pytz==2017.3
    PyYAML==3.12
    redis==2.10.6
    redo==1.6
    regex==2017.9.23
    requests==2.18.4
    requests-file==1.4.3
    requests-ftp==0.3.1
    requests-toolbelt==0.8.0
    robobrowser==0.5.3
    rsa==3.4.2
    s2sphere==0.2.4
    s3transfer==0.1.12
    scapy==2.3.2
    scikit-learn==0.19.0
    scipy==1.0.0rc1
    selenium==3.8.0
    service-identity==17.0.0
    sip==4.19.3
    six==1.11.0
    sklearn==0.0
    snowballstemmer==1.2.1
    sortedcontainers==1.5.9
    spacy==1.9.0
    speaklater==1.3
    Sphinx==1.3.5
    sphinx-rtd-theme==0.1.9
    splinter==0.7.4
    SQLAlchemy==1.2.0
    statsmodels==0.8.0
    TA-Lib==0.4.15
    tables==3.4.2
    tensorflow==1.3.0
    tensorflow-tensorboard==0.1.8
    termcolor==1.1.0
    texttable==0.8.4
    thinc==6.5.2
    toolz==0.8.2
    tornado==4.5.3
    tox==2.3.1
    tqdm==4.19.1.post1
    traitlets==4.3.2
    twisted==17.9.0
    txaio==2.8.2
    ujson==1.35
    update==0.4.4
    uritemplate==3.0.0
    urllib3==1.22
    virtualenv==14.0.6
    wcwidth==0.1.7
    websocket==0.2.1
    websocket-client==0.44.0
    Werkzeug==0.12.2
    winrandom-ctypes==1.0.3
    wrapt==1.10.11
    WTForms==2.1
    xxhash==0.6.1
    yarl==0.17.0
    zipline==1.1.1
    zope.interface==4.4.3
    Now that you know a little about me, let me tell you about the issue I am
    having:

Description of Issue
```
python.exe P:/telegram_botz/bot/examples_catalyst/buy_low_sell_high.py
[2018-01-16 17:10:31.163193] WARNING: run_algo: Catalyst is currently in ALPHA. It is going through rapid development and it is subject to errors. Please use carefully. We encourage you to report any issue on GitHub: https://github.com/enigmampc/catalyst/issues
[2018-01-16 17:10:34.177975] INFO: run_algo: running algo in paper-trading mode
[2018-01-16 17:15:02.071945] WARNING: Loader: Refusing to download new treasury data because a download succeeded at 2018-01-16 16:41:52.410097+00:00.
[2018-01-16 17:15:02.118821] INFO: exchange_algorithm: initialized trading algorithm in live mode
Traceback (most recent call last):
  File "P:/telegram_botz/bot/examples_catalyst/buy_low_sell_high.py", line 157, in <module>
    live_graph=True#,analyze_live=True
  File "E:\WARTER\python\python.v3_portable\App\lib\site-packages\catalyst\utils\run_algo.py", line 535, in run_algorithm
    stats_output=stats_output
  File "E:\WARTER\python\python.v3_portable\App\lib\site-packages\catalyst\utils\run_algo.py", line 314, in _run
    overwrite_sim_params=False,
  File "E:\WARTER\python\python.v3_portable\App\lib\site-packages\catalyst\exchange\exchange_algorithm.py", line 297, in run
    data, overwrite_sim_params
  File "E:\WARTER\python\python.v3_portable\App\lib\site-packages\catalyst\algorithm.py", line 724, in run
    for perf in self.get_generator():
  File "E:\WARTER\python\python.v3_portable\App\lib\site-packages\catalyst\exchange\exchange_algorithm.py", line 534, in get_generator
    self._init_trading_client()
  File "E:\WARTER\python\python.v3_portable\App\lib\site-packages\catalyst\exchange\exchange_algorithm.py", line 491, in _init_trading_client
    key='cumulative_performance',
  File "E:\WARTER\python\python.v3_portable\App\lib\site-packages\catalyst\exchange\utils\exchange_utils.py", line 297, in get_algo_object
    return pickle.load(handle)
  File "pandas/tslib.pyx", line 295, in pandas.tslib.Timestamp.__new__ (pandas\tslib.c:9203)
  File "pandas/tslib.pyx", line 1343, in pandas.tslib.convert_to_tsobject (pandas\tslib.c:25854)
TypeError: Cannot convert input to Timestamp
```

buy_low_sell_high.py
```
import talib
import pandas as pd
from logbook import Logger

from catalyst.api import (
    order,
    order_target_percent,
    symbol,
    record,
    get_open_orders,
)
from catalyst.exchange.utils.stats_utils import get_pretty_stats
from catalyst.utils.run_algo import run_algorithm

algo_namespace = 'buy_the_dip_live'
log = Logger('buy low sell high')


def initialize(context):
    log.info('initializing algo')
    context.ASSET_NAME = 'bnb_btc'
    context.asset = symbol(context.ASSET_NAME)

    context.TARGET_POSITIONS = 30
    context.PROFIT_TARGET = 0.1
    context.SLIPPAGE_ALLOWED = 0.02

    context.errors = []
    pass


def _handle_data(context, data):
    price = data.current(context.asset, 'price')
    log.info('got price {price}'.format(price=price))

    prices = data.history(
        context.asset,
        fields='price',
        bar_count=20,
        frequency='5T'
    )
    rsi = talib.RSI(prices.values, timeperiod=14)[-1]
    log.info('got rsi: {}'.format(rsi))

    # Buying more when RSI is low, this should lower our cost basis
    if rsi <= 30:
        buy_increment = 1
    elif rsi <= 40:
        buy_increment = 0.5
    elif rsi <= 70:
        buy_increment = 0.2
    else:
        buy_increment = 0.1

    cash = context.portfolio.cash
    log.info('base currency available: {cash}'.format(cash=cash))

    record(
        price=price,
        rsi=rsi,
    )

    orders = get_open_orders(context.asset)
    if orders:
        log.info('skipping bar until all open orders execute')
        return

    is_buy = False
    cost_basis = None
    if context.asset in context.portfolio.positions:
        position = context.portfolio.positions[context.asset]

        cost_basis = position.cost_basis
        log.info(
            'found {amount} positions with cost basis {cost_basis}'.format(
                amount=position.amount,
                cost_basis=cost_basis
            )
        )

        if position.amount >= context.TARGET_POSITIONS:
            log.info('reached positions target: {}'.format(position.amount))
            return

        if price < cost_basis:
            is_buy = True
        elif (position.amount > 0
              and price > cost_basis * (1 + context.PROFIT_TARGET)):
            profit = (price * position.amount) - (cost_basis * position.amount)
            log.info('closing position, taking profit: {}'.format(profit))
            order_target_percent(
                asset=context.asset,
                target=0,
                limit_price=price * (1 - context.SLIPPAGE_ALLOWED),
            )
        else:
            log.info('no buy or sell opportunity found')
    else:
        is_buy = True

    if is_buy:
        if buy_increment is None:
            log.info('the rsi is too high to consider buying {}'.format(rsi))
            return

        if price * buy_increment > cash:
            log.info('not enough base currency to consider buying')
            return

        log.info(
            'buying position cheaper than cost basis {} < {}'.format(
                price,
                cost_basis
            )
        )
        order(
            asset=context.asset,
            amount=buy_increment,
            limit_price=price * (1 + context.SLIPPAGE_ALLOWED)
        )


def handle_data(context, data):
    log.info('handling bar {}'.format(data.current_dt))
    # try:
    _handle_data(context, data)
    # except Exception as e:
    #     log.warn('aborting the bar on error {}'.format(e))
    #     context.errors.append(e)

    log.info('completed bar {}, total execution errors {}'.format(
        data.current_dt,
        len(context.errors)
    ))

    if len(context.errors) > 0:
        log.info('the errors:\n{}'.format(context.errors))


def analyze(context, stats):
    log.info('the daily stats:\n{}'.format(get_pretty_stats(stats)))
    pass


if __name__ == '__main__':
    live = True
    if live:
        run_algorithm(
            capital_base=0.001,
            initialize=initialize,
            handle_data=handle_data,
            analyze=analyze,
            exchange_name='binance',
            live=True,
            algo_namespace=algo_namespace,
            base_currency='btc',
            live_graph=True#,analyze_live=True
        )
    else:
        run_algorithm(
            capital_base=10000,
            data_frequency='daily',
            initialize=initialize,
            handle_data=handle_data,
            analyze=analyze,
            exchange_name='binance',
            algo_namespace='buy_and_hodl',
            base_currency='btc',
            start=pd.to_datetime('2018-01-01', utc=True),
            end=pd.to_datetime('2018-01-16', utc=True),
        )
# run_algorithm()

```

**Comment by fredfortier** [comment]
I just tested with the `buy_low_sell_high.py` algo in our examples folder and it works find in backtest and paper trading mode. This must have been a Python3 issue. We are now testing more extensively with Python3 and made several adjustments.

**Comment by fredfortier** [comment]
Fixed on 0.4.6 please re-open if you still notice an issue.
